lib/msal-common/src/client/DeviceCodeClient.ts

```diff
@@ -196,17 +197,23 @@ export class DeviceCodeClient extends BaseClient {
                 shrClaims: request.shrClaims,
                 sshKid: request.sshKid
             };
+
             const response = await this.executePostToTokenEndpoint(
                 this.authority.tokenEndpoint,
                 requestBody,
                 headers,
                 thumbprint);
 
-            if (response.body && response.body.error === Constants.AUTHORIZATION_PENDING) {
+            if (response.body && response.body.error) {
                 // user authorization is pending. Sleep for polling interval and try again
-                this.logger.info(response.body.error_description || "Authorization pending. Continue polling.");
-  
-                await TimeUtils.delay(pollingIntervalMilli);
+                if(response.body.error === Constants.AUTHORIZATION_PENDING) {
+                    this.logger.info(response.body.error_description || "Authorization pending. Continue polling.");
```

Comment:
What's the scenario where the OR condition of this log message gets hit?

Reply:
I guess to accommodate different server end points we may support. AAD always sends one type of error afaik.

Reply:
Can we just log the right side then? Not sure I understand why we need both

Reply:
This code hasn't changed in a while. I need to do some more experimentation to see why we need both before removing it to ensure we are not breaking anyone. Can we take this as a separate PBI?

Reply:
Done.